app/src/main/java/io/apicurio/registry/Version.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2021 Red Hat
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.apicurio.registry;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Properties;
+
+import javax.annotation.PostConstruct;
+import javax.enterprise.context.ApplicationScoped;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @author eric.wittmann@gmail.com
+ */
+@ApplicationScoped
+public class Version {
+
+    private static Logger logger = LoggerFactory.getLogger(Version.class);
+
+    private Properties properties;
+
+    @PostConstruct
+    public void load() {
+        this.properties = new Properties();
+        try (InputStream input = Version.class.getResourceAsStream("version.properties")) {
```

Comment:
I don't know if I missing something, but, why don't you get this information from the maven build?

Reply:
We are.  Because version.properties is in src/main/resources, it is a filtered resource.  Which means that the values in that file are set by the maven build when copying the file to `target` in prep to make the JAR.  So basically the version # and build date is set by maven when doing a build.

I guess I could have done the same thing via application.properties, eh?  Would that be better?  I basically copied this approach from Studio, which doesn't have an application.properties file. :)

Reply:
no no , it's perfectly fine. I wasn't aware of any version.properties file. if it works as you described I'm perfectly ok :)